lib/Runtime/Library/JavascriptRegularExpression.cpp

```diff
@@ -671,11 +671,37 @@ namespace Js
         Assert(!(callInfo.Flags & CallFlags_New));
         Assert(args.Info.Count > 0);
 
-        JavascriptRegExp* obj = GetJavascriptRegExp(args, _u("RegExp.prototype.toString"), scriptContext);
+        PCWSTR const varName = _u("RegExp.prototype.toString");
 
-        bool sourceOnly = false;
-        bool useFlagsProperty = scriptContext->GetConfig()->IsES6RegExPrototypePropertiesEnabled();
-        return obj->ToString(sourceOnly, useFlagsProperty);
+        const ScriptConfiguration* scriptConfig = scriptContext->GetConfig();
+
+        if (scriptConfig->IsES6RegExPrototypePropertiesEnabled())
+        {
+            RecyclableObject *thisObj = GetThisObject(args, varName, scriptContext);
+            JavascriptString* source = JavascriptConversion::ToString(
+                JavascriptOperators::GetProperty(thisObj, PropertyIds::source, scriptContext),
+                scriptContext);
+            JavascriptString* flags = JavascriptConversion::ToString(
+                JavascriptOperators::GetProperty(thisObj, PropertyIds::flags, scriptContext),
+                scriptContext);
+
+            CharCount length = source->GetLength() + flags->GetLength() + 2; // 2 for the two '/'s
+            CompoundString *const builder =
+                CompoundString::NewWithCharCapacity(length, scriptContext->GetLibrary());
+            builder->Append(_u('/'));
+            builder->Append(source);
+            builder->Append(_u('/'));
+            builder->Append(flags);
+            return builder;
+        }
+        else
+        {
+            JavascriptRegExp* obj = GetJavascriptRegExp(args, varName, scriptContext);
+
+            bool sourceOnly = false;
+            bool useFlagsProperty = scriptConfig->IsES6RegExPrototypePropertiesEnabled();
+            return obj->ToString(sourceOnly, useFlagsProperty);
```

Comment:
This isn't the only place. It is also used to generate the string for the `source` property. However, now that `EntryToString` handles the `flags` property itself, the part of `ToString` handling it is unreachable. I've removed that section in commit 154d829011cd3290c7ba15a60bad3df250d5936b.